src/model/inc_classifier.py

```diff
@@ -64,15 +66,18 @@ def increment(self, new_classes: list[int]):
                 self.active_units[new_classes] = 1
 
         if old_nclasses != new_nclasses:
+            self.mul *= old_nclasses/new_nclasses
             state_dict = self.classifier.state_dict()
             self.classifier = self.get_classifier(in_features, new_nclasses).to(device)
-            for name, param in self.classifier.named_parameters():
+            for name, param in self.classifier.parameters():
                 param.data[:old_nclasses] = state_dict[name]
         self.out_features = new_nclasses
         
 
     def forward(self, x):
-        out = self.classifier(x)
+        x = self.classifier[0](x)
+        x *= self.mul
+        out = self.classifier[1](x)
```

Comment:
There should be x = x*old_nclasses/new_nclasses + x_min * (1 - old_nclasses/new_nclasses) in line 79. Without the shift, it simply does not work because it only shrinks the domain within which the hills are defined. I wrote a proof demonstrating that the proposed scaling is valid. Please check our Overleaf and reach out to me if you have any questions.